x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlQueryRequest.java

```diff
@@ -65,17 +67,47 @@ public EsqlQueryRequest(StreamInput in) throws IOException {
     @Override
     public ActionRequestValidationException validate() {
         ActionRequestValidationException validationException = null;
+        if (Strings.hasText(esqlVersion) == false) {
+            // TODO: make this required
+            // "https://github.com/elastic/elasticsearch/issues/104890"
+            // validationException = addValidationError("[" + RequestXContent.ESQL_VERSION_FIELD + "] is required", validationException);
+        } else {
+            EsqlVersion version = EsqlVersion.parse(esqlVersion);
+            if (version == null) {
+                validationException = addValidationError(
+                    "[" + RequestXContent.ESQL_VERSION_FIELD + "] has invalid value [" + esqlVersion + "]",
+                    validationException
+                );
+            } else if (version == EsqlVersion.NIGHTLY && Build.current().isSnapshot() == false) {
```

Comment:
Thinking out loud - does the snapshot check belong here, or in the `EsqlVersion.parse` method?  I can see arguments for both.

Reply:
Hm, personally I do think it belongs here - `EsqlVersion` shouldn't need to worry about where we permit certain versions to be used IMHO, and why; also, this allows to keep `EsqlVersion.parse` simple: either the string is a valid version or it's not - otherwise we'd have to throw an exception containing the reason why parsing failed and catch it in `EsqlQueryRequest.validate()`. We might also parse versions coming from elsewhere in the future, and wouldn't want to prevent parsing of "nightly" just because we're not on a snapshot version.